src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -175,10 +175,10 @@ class FastSync(
     private def discardLastBlocks(startBlock: BigInt, blocksToDiscard: Int): Unit = {
       (startBlock to ((startBlock - blocksToDiscard) max 1) by -1).foreach { n =>
         blockchain.getBlockHeaderByNumber(n).foreach { headerToRemove =>
-          blockchain.removeBlock(headerToRemove.hash, saveParentAsBestBlock = false)
+          blockchain.removeBlock(headerToRemove.hash, withState = false)
         }
       }
-      blockchain.saveBestBlockNumber((startBlock - blocksToDiscard - 1) max 0)
+      appStateStorage.putBestBlockNumber((startBlock - blocksToDiscard - 1) max 0)
```

Comment:
I'm not quite sure about this change. Shouldn't we change `io.iohk.ethereum.domain.BlockchainImpl#bestKnownBlock` as well?

Reply:
I have actually totally gone around cache in case of fast sync, any node save just write it to storage, so best block is also written to storage when neccesssery.